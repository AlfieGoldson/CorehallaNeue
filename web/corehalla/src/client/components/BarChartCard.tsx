// Library imports
import React, { FC } from 'react';
import styled from 'styled-components';

// Components imports
import { Card } from './Card';
import { BarChart } from './Charts';
import { StatDesc, StatSmall } from './TextStyles';

interface BarChartStat {
    title: string;
    amount: number;
    max: number;
}

interface Props {
    title: string;
    stats: BarChartStat[];
}

const BarChartStatWrapper = styled.div`
    margin-bottom: 1rem;
    display: flex;
    flex-direction: column;

    & > div {
        margin-bottom: 0.25rem;
    }
`;

export const BarChartCard: FC<Props> = ({ title, stats }: Props) => {
    return (
        <Card title={title}>
            {stats.map(({ title: statTitle, amount, max }, i) => (
                <BarChartStatWrapper key={i}>
                    <div>
                        <StatSmall>{amount}</StatSmall>
                        <StatDesc>{statTitle}</StatDesc>
                    </div>
                    <BarChart width="100%" height="0.25rem" amount={(amount / max) * 100} />
                </BarChartStatWrapper>
            ))}
        </Card>
    );
};
