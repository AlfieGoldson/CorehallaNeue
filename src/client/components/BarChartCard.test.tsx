import React from 'react';
import { shallow } from 'enzyme';
import { BarChartCard } from './BarChartCard';

describe('BarChartCard', () => {
    it('should render correctly with no stats', () => {
        const component = shallow(<BarChartCard title="barchart" stats={[]} />);
        expect(component).toMatchSnapshot();
    });
    it('should render correctly with stats', () => {
        const component = shallow(<BarChartCard title="barchart" stats={[{ title: 'stats', amount: 30, max: 40 }]} />);
        expect(component).toMatchSnapshot();
    });
});
