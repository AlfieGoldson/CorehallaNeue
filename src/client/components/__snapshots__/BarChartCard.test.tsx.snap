// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BarChartCard should render correctly with no stats 1`] = `
<Component
  title="barchart"
/>
`;

exports[`BarChartCard should render correctly with stats 1`] = `
<Component
  title="barchart"
>
  <styled.div
    key="0"
  >
    <div>
      <styled.span>
        30
      </styled.span>
      <styled.span>
        stats
      </styled.span>
    </div>
    <Component
      amount={75}
      height="0.25rem"
      width="100%"
    />
  </styled.div>
</Component>
`;
